src/benchmarklib/abstract_table_generator.cpp

```diff
@@ -348,6 +327,76 @@ std::shared_ptr<BenchmarkConfig> AbstractTableGenerator::create_benchmark_config
   return std::make_shared<BenchmarkConfig>(config);
 }
 
+void AbstractTableGenerator::_create_chunk_indexes(
+    std::unordered_map<std::string, BenchmarkTableInfo>& table_info_by_name) {
+  Timer timer;
+  std::cout << "- Creating chunk indexes" << std::endl;
+  const auto& indexes_by_table = _indexes_by_table();
+  if (indexes_by_table.empty()) {
+    std::cout << "-  No indexes defined by benchmark" << std::endl;
+  }
+  for (const auto& [table_name, indexes] : indexes_by_table) {
+    const auto& table = table_info_by_name[table_name].table;
+
+    for (const auto& index_columns : indexes) {
+      auto column_ids = std::vector<ColumnID>{};
+
+      for (const auto& index_column : index_columns) {
+        column_ids.emplace_back(table->column_id_by_name(index_column));
+      }
+
+      std::cout << "-  Creating index on " << table_name << " [ ";
+      for (const auto& index_column : index_columns) {
+        std::cout << index_column << " ";
+      }
+      std::cout << "] " << std::flush;
+      Timer per_index_timer;
+
+      if (column_ids.size() == 1) {
+        table->create_index<GroupKeyIndex>(column_ids);
+      } else {
+        table->create_index<CompositeGroupKeyIndex>(column_ids);
+      }
+
+      std::cout << "(" << per_index_timer.lap_formatted() << ")" << std::endl;
+    }
+  }
+  metrics.chunk_index_duration = timer.lap();
+  std::cout << "- Creating chunk indexes done (" << format_duration(metrics.chunk_index_duration) << ")" << std::endl;
+}
+
+void AbstractTableGenerator::_create_table_indexes(
+    std::unordered_map<std::string, BenchmarkTableInfo>& table_info_by_name) {
+  Timer timer;
+  std::cout << "- Creating table indexes" << std::endl;
+  const auto& indexes_by_table = _indexes_by_table();
+  if (indexes_by_table.empty()) {
+    std::cout << "-  No indexes defined by benchmark" << std::endl;
+  }
+  for (const auto& [table_name, indexes] : indexes_by_table) {
+    const auto& table = table_info_by_name[table_name].table;
+
+    auto chunk_ids = std::vector<ChunkID>(table->chunk_count());
+    std::iota(chunk_ids.begin(), chunk_ids.end(), ChunkID{0});
+    for (const auto& index_columns : indexes) {
+      if (index_columns.size() > 1) {
+        std::cout << "-  Requested multi-column index is stored as multiple single-column indexes." << std::endl;
```

Comment:
>We lack the capbility to use multiple indexes separately in the query execution (it's possible, but I don't see an efficient way of doing it in Hyrise).

Do you mean using multiple indexes separately during the execution of an individual operator? Using multiple indexes separately in the query execution across multiple operators shouldn't be an issue.

> The current solution here would lead to creating two indexes in tables with multiple key columns

This assumes that indexes are created for primary key columns, right? Based on the current state of Hyrise, we don't have such automation and the indexes to be created are specified in the corresponding TableGenerator.

> but I wouldn't do it for automatically created indexes on primary key columns.

But since we don't have this automation yet and we also don't support creating indexes via SQL statements, we cannot introduce a case distinction here.

Failing loudly when someone tries to create a milti column index could be an alternative here.

Reply:
When generating the TPCH benchmark multiple multi-column indexes are generated, so failing loudly would lead to other problems :/

Reply:
Right, but if we don't support multi-column indexes, we can adapt the index definitions accordingly so that we only create single-column indexes in our benchmarks then.

Pinging @Bouncner since there are some open questions above.